lumigator/jobs/inference/model_clients.py

```diff
@@ -151,6 +152,51 @@ def _chat_completion(
 class HuggingFaceModelClient(BaseModelClient):
     def __init__(self, config: InferenceJobConfig):
         self._pipeline = pipeline(**config.hf_pipeline.model_dump())
+        self._set_tokenizer_max_length()
+
+    def _set_tokenizer_max_length(self):
+        """Set the tokenizer's model_max_length if it's currently the default very large integer.
+        Checks various possible max_length parameters which varies on model architecture.
+        """
+        config = self._pipeline.model.config
+        logger.info(f"Initial model_max_length {self._pipeline.tokenizer.model_max_length}")
```

Comment:
In my opinion, don't say "initial ...". It kind of implies that you will definitely change it. You can also omit this log in favor of the other one, beneath.

```suggestion
        logger.info(f"The maximum number of input tokens is set to {self._pipeline.tokenizer.model_max_length}")
```

Reply:
Keeping this message, re-worded it.
Removing the one below (both will be the same value if `self._pipeline.tokenizer.model_max_length != VERY_LARGE_INTEGER`